parquet-column/src/main/java/parquet/filter2/recordlevel/IncrementallyUpdatedFilterPredicate.java

```diff
@@ -61,6 +64,24 @@
 
     private boolean result = false;
     private boolean isKnown = false;
+    private BitSet dictionaryResults;
+
+    /** Populates array of BitSet based on whether values pass filter */
+    void setDictionary(Dictionary dictionary) {
+      dictionaryResults = new BitSet(dictionary.getMaxId()+1);
+      for (int i = 0, end = dictionary.getMaxId()+1; i < end; i++) {
+        dictionaryResults.set(i, update(dictionary, i));
+      }
+    }
+
+    /** Update based on element from dictionary */
+    void updateFromDictionary(int dictionaryId) {
+      setResult(dictionaryResults.get(dictionaryId));
+    }
+
+    /** Does the element in the dictionary pass the filter */
+    abstract protected boolean update(Dictionary dictionary, int dictionaryId);
```

Comment:
I think there may be some confusion because I reused the update method name for this method, which is silly.
   /** Does the element in the dictionary pass the filter */
    abstract protected boolean update(Dictionary dictionary, int dictionaryId);
I will change it. It does not update anything, it is only called from setDictionary and is used to evaluate the predicate so that the result can be stored in the BitMap. I will change the name to evaluateFilter, or something simpler. 

The FilteringPrimitiveConverter then calls updateFromDictionary to make the update, which just accesses the value in the bitmap.